src/platform/android/AndroidChipPlatform-JNI.cpp

```diff
@@ -255,3 +257,15 @@ static bool JavaBytesToUUID(JNIEnv * env, jbyteArray value, chip::Ble::ChipBleUU
     return result;
 }
 #endif
+
+// for CommissionableDataProvider
+JNI_METHOD(jboolean, updateCommissionableDataProviderData)
+(JNIEnv * env, jclass self, jstring spake2pVerifierBase64, jstring Spake2pSaltBase64, jint spake2pIterationCount,
+ jlong setupPasscode, jint discriminator)
+{
+    chip::DeviceLayer::StackLock lock;
+    CHIP_ERROR err = CommissionableDataProviderMgrImpl().Update(env, spake2pVerifierBase64, Spake2pSaltBase64,
```

Comment:
On error, error information (code, maybe location) is lost because this returns bool sucess/failure.

How about we log the error in case of err != CHIP_NO_ERROR and in that way the logs would show what actually failed more than 'Update failed' that we can get with a bool.

Reply:
CommissionableDataProviderImpl::Update already logs everything, is that OK?

Reply:
I believe in "return codes should never be discarded", so when you get an error code, you have the option to handle it - generally pass it along or log it or something in between (e.g. handle some errors, log  others). This one drops the error data and converts it to a bool. This is insufficient to just rely that lower levels log things.

Reply:
OK, log added